### ISSUE TITLE
Trivy Overloaded When Multiple Projects Are Uploaded Without Delay

### ISSUE THREAD

**Issue by LaVibeX** [DependencyTrack/dependency-track#4453]
### Current Behavior

When uploading 200 projects sequentially via the API without introducing any delay, the analyzers successfully assign vulnerabilities to the components, but Trivy becomes unresponsive and fails to process further.

Adding a reasonable delay between uploads (e.g., 50 seconds in my case) allows Trivy to react and function properly.

### Steps to Reproduce

1. Sequentially upload 200 projects via the API, waiting for a server response after each upload.
2. Observe that vulnerabilities are assigned to components initially, but Trivy becomes stuck after some time.
3. Retry with a delay (e.g., 50 seconds) between uploads.
4. Observe that Trivy processes all projects successfully.

### Expected Behavior

Trivy should handle sequential project uploads without requiring an artificial delay.

Actual Behavior:
Trivy becomes stuck after processing a certain number of sequential project uploads, despite waiting for a server response for each upload.

### Dependency-Track Version

4.12.2

### Dependency-Track Distribution

Executable WAR

### Database Server

PostgreSQL

### Browser

Mozilla Firefox

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
When you say:

> Trivy becomes unresponsive and fails to process further

Do you mean the Trivy integration, or specifically Trivy itself? As in we are likely sending too many requests? Is it getting stuck or is it just slowing down?

**Comment by LaVibeX** [comment]
Hi @nscuro,

I was referring to the Trivy integration as a whole, not Trivy itself. Actually, after observing the behavior more closely, I noticed that Trivy doesn’t get stuck but slows down significantly compared to other analyzers.

For example, while most analyzers finish processing in seconds, Trivy took over 10 minutes or more per project in some cases. This happens when a lot of sequential requests are sent.

**Comment by valentijnscholten** [comment]
So we are the delays. Inside Dependency Track or inside Trivy? Are you running a self hosted Trivy?

**Comment by LaVibeX** [comment]
Yes, I’m running a self-hosted Trivy instance. What I’ve observed is that vulnerabilities from other analyzers appear significantly faster in Dependency Track Project than those from Trivy. How can I know from where the delays are coming?

**Comment by nscuro** [comment]
Given the current threading model, uploading 200 BOMs in short succession could lead to all worker threads performing Trivy analysis at the same time. On a system with a 4 core CPU you have 16 worker threads per default (`workerThreadCount = cpuCoreCount * 4`).

I have never personally tested if Trivy has any limitations on that front. But of course we can't exclude our analyzer being the culprit. 

@fnxpt Did you encounter this before in your usage?
